android/app/src/main/java/app/organicmaps/routing/NavigationService.java

```diff
@@ -199,21 +210,31 @@ public void onLocationUpdated(@NonNull Location location)
     if (!routingController.isNavigating())
       return;
 
-    // Voice the turn notification first.
+    // TODO: consider to create callback mechanism to transfer 'ROUTE_IS_FINISHED' event from
+    // the core to the platform code (https://github.com/organicmaps/organicmaps/issues/3589),
+    // because calling the native method 'nativeIsRouteFinished'
+    // too often can result in poor UI performance.
+    if (Framework.nativeIsRouteFinished())
```

Comment:
Good idea. Maybe create an issue?
cc @vng

Reply:
Please read the comment... https://github.com/organicmaps/organicmaps/issues/3589